contracts/Builders/SimpleBuilder.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../interfaces/IProvider.sol";
+import "../interfaces/ILockDealNFT.sol";
+import "@openzeppelin/contracts/token/ERC721/utils/ERC721Holder.sol";
+import "../util/CalcUtils.sol";
+
+/// @title SimpleBuilder contract
+/// @notice Implements a contract for building refund simple providers
+contract SimpleBuilder is ERC721Holder {
+    using CalcUtils for uint256;
+    ILockDealNFT public lockDealNFT;
+    IProvider public refundProvider;
+    IProvider public simpleProvider;
+    IProvider public collateralProvider;
+
+    constructor(ILockDealNFT _nft, IProvider _refund, IProvider _simpleProvider, IProvider _collateral) {
+        lockDealNFT = _nft;
+        refundProvider = _refund;
+        simpleProvider = _simpleProvider;
+        collateralProvider = _collateral;
+    }
+
+    struct UserPool {
+        address user;
+        uint256 amount;
+    }
+
+    /// @param userPools - array of user pools
+    /// @param addressParams[0] = token
+    /// @param addressParams[1] = mainCoin
+    /// @param params[0] = collateral params, [0] start amount, [1] finish time
+    function buildRefundSimple(
+        UserPool[] calldata userPools,
+        address[] calldata addressParams,
+        uint256[] calldata params
+    ) public {
+        //TODO require
+        require(userPools.length > 1, "invalid userPools length");
+        require(addressParams.length > 1, "invalid addressParams length");
+        require(params.length > 1, "invalid params length");
+        uint256 firstRefundPoolId = lockDealNFT.totalSupply();
+        (uint256 lastRefundPoolId, uint256 totalAmount) = _createRefundPools(addressParams, userPools);
+        address mainCoin = addressParams[1];
+        uint256 collateralPoolId = _createCollateralProvider(mainCoin, lastRefundPoolId, params);
+        // register refund pools
+        uint256 mainCoinAmount = params[0];
+        uint256 rateToWei = mainCoinAmount.calcRate(totalAmount);
+        _registerRefundPools(collateralPoolId, firstRefundPoolId, lastRefundPoolId, rateToWei);
+    }
+
+    function _createRefundPools(
+        address[] calldata addressParams,
+        UserPool[] calldata userPools
+    ) internal returns (uint256 lastPoolId, uint256 totalAmount) {
+        address token = addressParams[0];
+        uint256 length = userPools.length;
+        uint256[] memory params = new uint256[](simpleProvider.currentParamsTargetLenght());
+        for (uint256 i = 0; i < length; ++i) {
+            uint256 userAmount = userPools[i].amount;
+            address user = userPools[i].user;
+            // mint refund pools for users
+            lockDealNFT.mintAndTransfer(user, token, msg.sender, userAmount, refundProvider);
```

Comment:
Can we use mintAndTransfer only 1 time in the cycle?
we can do mint + copy valut Id